subsys/debug/cpu_load/cpu_load.c

```diff
@@ -0,0 +1,292 @@
+		     nrf_rtc_event_address_get(NRF_RTC1, NRF_RTC_EVENT_TICK),
+		     nrfx_timer_task_address_get(&timer, NRF_TIMER_TASK_COUNT));
+		nrf_rtc_event_enable(NRF_RTC1, NRF_RTC_INT_TICK_MASK);
+	}
+
+	err = ppi_alloc(&ch_sleep,
+		       nrf_power_event_address_get(NRF_POWER,
+						   NRF_POWER_EVENT_SLEEPENTER));
+	if (err != NRFX_SUCCESS) {
+		ppi_cleanup(ch_tick, CH_INVALID, CH_INVALID);
+		return -ENODEV;
+	}
+
+	err = ppi_alloc(&ch_wakeup,
+		       nrf_power_event_address_get(NRF_POWER,
+						   NRF_POWER_EVENT_SLEEPEXIT));
+	if (err != NRFX_SUCCESS) {
+		ppi_cleanup(ch_tick, ch_sleep, CH_INVALID);
+		return -ENODEV;
+	}
+
+	err = nrfx_timer_init(&timer, &config, timer_handler);
+	if (err != NRFX_SUCCESS) {
+		ppi_cleanup(ch_tick, ch_sleep, ch_wakeup);
+		return -EBUSY;
+	}
+
+	nrfx_gppi_channel_endpoints_setup(ch_sleep,
+		  nrf_power_event_address_get(NRF_POWER,
+					      NRF_POWER_EVENT_SLEEPENTER),
+		  nrfx_timer_task_address_get(&timer, NRF_TIMER_TASK_START));
+	nrfx_gppi_channel_endpoints_setup(ch_wakeup,
+		  nrf_power_event_address_get(NRF_POWER,
+					      NRF_POWER_EVENT_SLEEPEXIT),
+		  nrfx_timer_task_address_get(&timer, NRF_TIMER_TASK_STOP));
+
+	/* In case of DPPI event can only be assigned to a single channel. In
+	 * that case, cpu load can still subscribe to the channel but should
+	 * not control it. It may result in cpu load not working. User must
+	 * take care of that.
+	 */
+	nrfx_gppi_channels_enable((IS_CH_SHARED(ch_sleep) ? 0 : BIT(ch_sleep)) |
+				(IS_CH_SHARED(ch_wakeup) ? 0 : BIT(ch_wakeup)) |
+				(IS_CH_SHARED(ch_tick) ? 0 : BIT(ch_tick)));
+
+	cpu_load_reset();
+
+	if (IS_ENABLED(CONFIG_CPU_LOAD_LOG_PERIODIC)) {
+		ret = cpu_load_log_init();
+	}
+
+	ready = true;
+
+	return ret;
+}
+
+void cpu_load_reset(void)
+{
+	nrfx_timer_clear(&timer);
+	cycle_ref = k_cycle_get_32();
+}
+
+static u32_t sleep_ticks_to_us(u32_t ticks)
+{
+	return IS_ENABLED(CONFIG_CPU_LOAD_ALIGNED_CLOCKS) ?
+	   (u32_t)(((u64_t)ticks * 1000000) / CONFIG_SYS_CLOCK_TICKS_PER_SEC) :
+	   ticks;
+}
+
+u32_t cpu_load_get(void)
+{
+	u32_t sleep_us;
+	u32_t total_cyc;
+	u64_t total_us;
+	u64_t load;
+
+	sleep_us = sleep_ticks_to_us(nrfx_timer_capture(&timer, 0));
+	total_cyc = k_cycle_get_32() - cycle_ref;
+
+	total_us = ((u64_t)total_cyc * 1000000)/
+				CONFIG_SYS_CLOCK_TICKS_PER_SEC;
+
+	__ASSERT(total_us < UINT32_MAX, "Measurement is limited.");
+
+	/* Because of different clock sources for system clock and TIMER it
+	 * is possible that sleep time is bigger than total measured time.
+	 *
+	 * Note that for simplicity reasons module does not handle TIMER
+	 * overflow.
+	 */
+	load = (total_us > (u64_t)sleep_us) ?
+			(100000 * (total_us - (u64_t)sleep_us)) / total_us : 0;
+
+	return (u32_t)load;
+}
+
+static int cmd_cpu_load_get(const struct shell *shell, size_t argc, char **argv)
+{
+	u32_t load = cpu_load_get();
+	u32_t percent = load / 1000;
+	u32_t fraction = load % 1000;
+
+	if (!ready) {
```

Comment:
Last one detail. It's a bit late for this check here. `cpu_load_get()` has been already called.